stable_baselines3/common/envs/identity_env.py

```diff
@@ -1,14 +1,16 @@
-from typing import Optional, Union
+from typing import Any, Dict, Generic, Optional, Tuple, TypeVar, Union
 
+import gym
 import numpy as np
-from gym import Env, Space
-from gym.spaces import Box, Discrete, MultiBinary, MultiDiscrete
+from gym import spaces
 
 from stable_baselines3.common.type_aliases import GymObs, GymStepReturn
 
+T = TypeVar("T", int, np.ndarray)
 
-class IdentityEnv(Env):
-    def __init__(self, dim: Optional[int] = None, space: Optional[Space] = None, ep_length: int = 100):
+
+class IdentityEnv(gym.Env, Generic[T]):
```

Comment:
can we still use it the same way as before or is that a breaking change?

Reply:
Yes. It's just for type checking.

Reply:
Yes. It's just for type checking.

Reply:
To explain a little more:

With the previous typing, it was allowed for an instance to have the following typings:

```python
def reset(self) -> int:
def step(self, action: int) -> Tuple[int, ..., ..., ...]:
```

```python
def reset(self) -> int:
def step(self, action: int) -> Tuple[np.ndarray, ..., ..., ...]:
```

```python
def reset(self) -> int:
def step(self, action: np.ndarray) -> Tuple[int, ..., ..., ...]:
```

```python
def reset(self) -> int:
def step(self, action: np.ndarray) -> Tuple[np.ndarray, ..., ..., ...]:
```

```python
def reset(self) -> np.ndarray:
def step(self, action: int) -> Tuple[int, ..., ..., ...]:
```

```python
def reset(self) -> np.ndarray:
def step(self, action: int) -> Tuple[np.ndarray, ..., ..., ...]:
```

```python
def reset(self) -> np.ndarray:
def step(self, action: np.ndarray) -> Tuple[int, ..., ..., ...]:
```

```python
def reset(self) -> np.ndarray:
def step(self, action: np.ndarray) -> Tuple[np.ndarray, ..., ..., ...]:
```

From now on, the allowed typings are:

```python
def reset(self) -> int:
def step(self, action: int) -> Tuple[int, ..., ..., ...]:
```

```python
def reset(self) -> np.ndarray:
def step(self, action: np.ndarray) -> Tuple[np.ndarray, ..., ..., ...]:
```

After the switch to gym v0.26, there will be a lot of change of this kind to introduce.

Reply:
alright ;)